### PR TITLE
feat(create): enable git clone with submodels

### LINKED ISSUE #1159: sasjs create should eject docs submodule
Where a repo contains a docs submodule (such as [react](https://github.com/sasjs/react-seed-app)) the content should be pulled, and the submodule link broken as part of `sasjs create` execution

For react, the submodule folder is /public/docs 
For angular and the template_jobs repos, it's just `/docs`

### PR DESCRIPTION
**Pull request by YuryShkoda**
## Issue

Closes #1159 

## Intent

- On `sasjs create` command using SASjs seed apps a docs submodels also need to be cloned.

## Implementation

- Add `--recurse-submodules` attribute to `git clone` command within `createApp` utility.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [x] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

**Review by medjedovicm — CHANGES_REQUESTED** [review]
Removing .git and .gitmodules

**Comment by YuryShkoda** [comment]
> Removing .git and .gitmodules

fixed

**Review by medjedovicm — APPROVED** [review]

**Review by allanbowe — COMMENTED** [review]
Some comments would be nice to see!

File: src/utils/utils.ts [hunk 1]
```diff
@@ -90,24 +90,44 @@ function createApp(
...
   const gitBranch = repoUrl.includes('template_sasonly') ? 'master' : 'main'
 
+  const gitVersion: string = (shelljs
+    .exec(`git version`, { silent: true })
+    .stdout.match(/(?<!git version)\d+\.\d+\.\d+/) || '')[0].replace(/\./g, '')
+
   shelljs.exec(
-    `cd "${folderPath}" && git clone --depth 1 -b ${gitBranch} ${repoUrl} .`,
+    `cd "${folderPath}" && git clone --recurse${
+      parseInt(gitVersion) > 2130 ? '-submodules' : ''
```
Comment (allanbowe) [inline src/utils/utils.ts:99]: A comment to explain why this section is necessary would be good

Comment (YuryShkoda) [reply src/utils/utils.ts:99]: good point

Comment (YuryShkoda) [reply src/utils/utils.ts:100]: done

**Review by medjedovicm — APPROVED** [review]

**Pull request merged**